### ISSUE TITLE
When loaded a saved model from disk, predict is missing

### ISSUE THREAD

**Issue by Howard-ll** [tensorflow/decision-forests#25]
Hi

- Issue symptom
  When loaded a saved model from disk, predict is missing

- How to reproduce
  1) Create a model
  2) Save it to disk
  3) load it from disk
  4) call model.predict  (If I do like model(X, training=False), it works well)

        # model creation
        model = tfdf.keras.RandomForestModel(num_trees=n_trees, max_depth=depth, min_examples=1)
        model.fit(x=x_selected, y=y)

        # model save
        path = './random_forest'
        os.makedirs(path, exist_ok=True)
        file_name = tempfile.TemporaryDirectory(dir=path).name
        model.save(file_name)
        # model load
        loaded_model = tf.saved_model.load(file_name)
        # inference
        score = loaded_model.predict(x_selected)

AttributeError: '_UserObject' object has no attribute 'predict'

Thanks!

**Comment by achoum** [comment]
**TLDR**

Try replacing

```python
loaded_model = tf.saved_model.load(file_name)
```

by 

```python
loaded_model = tf.keras.models.load_model(file_name)
```

**Details**

The `predict` method is not part of the TensorFlow SavedModel object itself.
Instead, it is created and restored by Keras (which is build on top of TensorFlow).
If you work work with Keras models, it is best to load them with `tf.keras.models.load_model`.

**Comment by Howard-ll** [comment]
> ```python
> tf.keras.models.load_model
> ```

Thanks! You are right.
